webapi/Controllers/ChatHistoryController.cs

```diff
@@ -260,4 +268,103 @@ public async Task<IActionResult> GetAllChatSessionsAsync(string userId)
 
         return this.NotFound($"No chat session found for chat id '{chatId}'.");
     }
+
+    /// <summary>
+    /// Delete a chat session.
+    /// </summary>
+    /// <param name="userId">Unique Id of user who initiated the request.</param>
+    /// <param name="sessionId">The chat id.</param>
+    [HttpDelete]
+    [Route("chatSession/{sessionId:guid}")]
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [Authorize(Policy = AuthPolicyName.RequireChatParticipant)]
+    public async Task<IActionResult> DeleteChatSessionAsync(
+        [FromServices] IHubContext<MessageRelayHub> messageRelayHubContext,
+        [FromServices] IAuthInfo authInfo,
+        Guid sessionId,
+        CancellationToken cancellationToken)
+    {
+        var chatId = sessionId.ToString();
+        ChatSession? chatToDelete = null;
+        try
+        {
+            // Make sure the chat session exists
+            chatToDelete = await this._sessionRepository.FindByIdAsync(chatId);
+        }
+        catch (KeyNotFoundException)
+        {
+            return this.NotFound($"No chat session found for chat id '{chatId}'.");
+        }
+
+        // Delete any resources associated with the chat session.
+        var deleteResourcesResult = await this.DeleteChatResourcesAsync(sessionId, cancellationToken) as StatusCodeResult;
+        if (deleteResourcesResult?.StatusCode != 204)
+        {
+            return this.StatusCode(500, $"Failed to delete resources for chat id '{chatId}'.");
```

Comment:
If this happens, you'll also want to log it, along with the status code

Reply:
We log individual resource failures in DeleteChatResources
![image](https://github.com/microsoft/chat-copilot/assets/125500434/7865b3df-e06b-4a35-91bc-c22393e11ee3)